### PR TITLE
_389-ds-base: 2.4.3 -> 2.4.5

### PR DESCRIPTION
**Pull request by h7x4**
## Description of changes

Comparison: https://github.com/389ds/389-ds-base/compare/389-ds-base-2.4.3...389-ds-base-2.4.5

Release notes:

- https://www.port389.org/docs/389ds/releases/release-2-4-4.html
- https://www.port389.org/docs/389ds/releases/release-2-4-5.html

maintainers: @ners 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by kirillrdy — APPROVED** [review]

**Comment by billksun** [comment]
Hi, first of all, I want to apologize for posting an off-topic message here. I need help with setting up 389ds on NixOS, but there doesn't seem to be many that have that experience on the [discourse forum](https://discourse.nixos.org/t/setting-up-389-ds-base/39269), so I thought I might try asking the packagers.

**Comment by h7x4** [comment]
No worries, but let's continue the discussion over at discourse :)

File: pkgs/servers/ldap/389/default.nix [hunk 1]
```diff
@@ -37,20 +37,20 @@
...
-  version = "2.4.3";
+  version = "2.4.5";
 
   src = fetchFromGitHub {
     owner = "389ds";
     repo = pname;
     rev = "${pname}-${version}";
-    hash = "sha256-bUjL1fjzyrq9jjpB/xbRCAISiPBwrlXbbDqT0aLOVOc=";
+    hash = "sha256-12JCd2R00L0T5EPUNO/Aw2HRID+z2krNQ09RSX9Qkj8=";
   };
```
Comment (risicle) [inline pkgs/servers/ldap/389/default.nix:48]:
```suggestion
  patches = [
    (fetchpatch {
      name = "fix-32bit.patch";
      url = "https://github.com/389ds/389-ds-base/commit/1fe029c495cc9f069c989cfbb09d449a078c56e2.patch";
      hash = "sha256-b0HSaDjuEUKERIXKg8np+lZDdZNmrCTAXybJzF+0hq0=";
    })
  ];
```

This patch is needed to fix 32bit systems accidentally broken in 2.4.4.

**Comment by risicle** [comment]
This bump is needed to fix build against glibc 2.39 #287594

**Comment by arcnmx** [comment]
:+1:

This should really be merged (without the patch would still unblock x86_64 at least) but now also needs to revert https://github.com/NixOS/nixpkgs/pull/287594/commits/1c003da73c7ca9e997a85781261ecd379eb4dd9a to unmark the package as broken.

**Comment by benley** [comment]
Thanks for this PR - we're still a few versions behind, but at least the package can be built again now!

**Pull request merged**